libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/useFiltersWithAddedPlaceholder.tsx

```diff
@@ -69,18 +72,31 @@ export function useFiltersWithAddedPlaceholder(filters: FilterContextItem[]): [
     },
 ] {
     const dispatch = useDashboardDispatch();
+    const selectedFilterIndex = useDashboardSelector(selectSelectedFilterIndex);
+
     const [[dateFilter], attributeFilters] = partition(filters, isDashboardDateFilter);
     const [addedAttributeFilter, setAddedAttributeFilter] = useState<
         FilterBarAttributeFilterPlaceholder | undefined
     >();
 
-    const addAttributeFilterPlaceholder = useCallback(function (index: number) {
-        setAddedAttributeFilter({ type: "attributeFilterPlaceholder", filterIndex: index });
-    }, []);
+    const addAttributeFilterPlaceholder = useCallback(
+        function (index: number) {
+            dispatch(uiActions.selectFilterIndex(index));
+        },
+        [dispatch],
+    );
 
     const clearAddedFilter = useCallback(() => {
+        dispatch(uiActions.clearFilterIndexSelection());
+    }, [dispatch]);
+
+    useEffect(() => {
```

Comment:
This does not need to be an effect, rather it should be a memo. If there is no selected filter, it would return undefined, if there is, return the appropriate placeholder.